deadbeef: update to 1.9.1

WhiredPlanck opened the pull request:
<!-- For description on topic creation and maintenance, please refer to [this Wiki article](https://wiki.aosc.io/developer/packaging/topic-based-maintenance-guideline/). -->

Topic Description
-----------------

This PR is to update `deadbeef` to 1.9.1, and introduce its new dependency `libdispatch`, which will replace old and orphaned `libblocksruntime` and `libdispatch-objc2`.

Package(s) Affected
-------------------

- `deadbeef` v1.9.1 (update)
- `libdispatch` v5.6.1 (new)
- `libblocksruntime` v0.4.1-2 (drop, replaced by `libdispatch`)
- `libdispatch-objc2` v20140226-3 (drop, replaced by `libdispatch`)

Security Update?
----------------

<!-- If this topic contains security update(s), please uncomment "Yes,"
     mark with the `security` and `priority` labels, and make sure to mark your commits to relevant issue numbers.

     Please see GitHub's documentation on "Linking a pull request to an issue":

     https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue -->

<!-- Yes - Issue Number: ISSUENUMBER -->
No

<!-- Please uncomment the "Build Order" section if your topic affects more than one package. -->

Build Order
-----------

`libdispatch` then `deadbeef`.

Test Build(s) Done
------------------

**Primary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->    
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

**Secondary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

<!-- Maintainers should review file changes and, if the build script(s) affected complies with the
     [Package Styling Manual](https://wiki.aosc.io/developer/packaging/package-styling-manual/), please
     add `lgtm` label to this issue. -->

<!-- Maintainers and users may now test the packages in this topic and, once user/maintainer feedback indicates that the update(s) work as expected and find its quality satisfactory,
     another maintainer may now review this pull request and mark it as Approved. After which, the maintainer will build affected package(s) and upload them to the `stable` repository. -->

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

**Secondary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

<!-- TODO: CI to auto-fill architectural progress. -->

WhiredPlanck committed:
deadbeef: update to 1.9.1

WhiredPlanck committed:
libdispatch: new, 5.6.1

WhiredPlanck committed:
deadbeef: replace libdispatch-ojbc2 with libdispatch

They are different libraries and libdispatch-ojbc2 doesn't provide
libblocksruntime, which DeadBeef requires.

The original, standalone libblocksruntime is old and outdated and should be
replaced by libdispatch, now maintained as a swift corelibs project.

Reference: https://apple.github.io/swift-corelibs-libdispatch/.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

WhiredPlanck committed:
deadbeef: rewrap AFTER parameters

WhiredPlanck committed:
deadbeef: reenable shadow build

Test build succeeded with shadow build.

WhiredPlanck committed:
deadbeef: drop unneeded appended flags

Test build succeeded without these flags.

WhiredPlanck committed:
libdispatch: mark PKGBREAK

Add libblocksruntime and libdispatch-objc2 to it.

WhiredPlanck committed:
libblocksruntime: drop, orphaned

Standalone libblocksruntime is old, outdated, orphaned and not required by any package anymore, it should be replaced by libdispatch.

WhiredPlanck committed:
libdispatch: add libdispatch-objc2 to PKGCONFL, PKGREP and PKGPROV

Both libdispatch and libdispatch-objc2 provide libdispatch.so. Nowadays the latter is old and outdated, so it should be replaced by libdispatch.

WhiredPlanck committed:
libdispatch-objc2: drop, orphaned

Standalone libdispatch is old, outdated, orphaned and not required by any package anymore, it should be replaced by libdispatch.

WhiredPlanck commented:
LGTM on AMD64
![图片](https://user-images.githubusercontent.com/47623588/168819442-efb0c58e-f537-4bd4-b0f0-53603322e7b1.png)

Test built packages are available in topic repo except RISC-V architecture (I cannot connect to build machine now).

WhiredPlanck commented:
/build

WhiredPlanck commented:
/build deadbeef

MingcongBai commented on extra-libs/libdispatch/autobuild/defines:
```diff
@@ -0,0 +1,12 @@
+PKGDEP="glibc"
+BUILDDEP="llvm"
+
+PKGBREAK="libblocksruntime>=0.4.1-2 libdispatch-objc2>=20140226-3"
```
Should be `<=`.

MingcongBai commented on extra-libs/libdispatch/autobuild/defines:
```diff
@@ -0,0 +1,12 @@
+PKGNAME=libdispatch
+PKGSEC=libs
+PKGDES="A library provides support for concurrency on multicore hardware"
```
Support library for concurrent code execution on multi-core hardware.

Adopted from [upstream homepage](https://apple.github.io/swift-corelibs-libdispatch/).

WhiredPlanck committed:
libdispatch: polish package description

Also fix typos in PKGBREAK: this package should to break those packages whose versions are LESS than the specified versions.
